src/main/src/main/java/org/geoserver/security/decorators/DefaultSecureDataFactory.java

```diff
@@ -114,14 +115,16 @@ public Object secure(Object object, WrapperPolicy policy) {
         } else if (FeatureIterator.class.isAssignableFrom(clazz)) {
             return new SecuredFeatureIterator((FeatureIterator) object); 
         }
-        
+
         // try coverage readers and formats
         if(GridCoverage2DReader.class.isAssignableFrom(clazz)) {
             return new SecuredGridCoverage2DReader((GridCoverage2DReader) object, policy);
+        } else if(StructuredGridCoverage2DReader.class.isAssignableFrom(clazz)) {
```

Comment:
Given that:

{code}
public interface StructuredGridCoverage2DReader extends GridCoverage2DReader {
{code}

Should't you test the structured one before testing its super-interface?